### PR TITLE
Fix/neural docs

### PR DESCRIPTION
**Pull request by michalk8**
Fixes the `ott.neural` docs. Also fixes some minor typos and not correctly forwarding argument in `ott.neural.losses.monge_gap`.
@marcocuturi I left 3 small todos to update some of the description in the docs.

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -90,6 +90,8 @@ Packages
   When the problem is *not* convex, which is the case for most other uses of
   this toolbox, the initialization can play a decisive role to reach a useful
   solution.
+- :mod:`ott.neural`  .. TODO(marcocuturi): add some nice text here please.
```
Comment (marcocuturi) [inline docs/index.rst:93]: provides tools to parameterize optimal transport maps, couplings or conditional probabilities as neural networks.

File: src/ott/geometry/distrib_costs.py [hunk 2]
```diff
@@ -47,22 +50,21 @@ def __init__(
...
-    from ott.solvers.linear import univariate
     super().__init__()
 
     self.ground_cost = (
         costs.SqEuclidean() if ground_cost is None else ground_cost
     )
-
     self._solver = univariate.UnivariateSolver() if solver is None else solver
     self._kwargs_solve = kwargs
+    self._kwargs_solve["return_transport"] = False
```
Comment (marcocuturi) [inline src/ott/geometry/distrib_costs.py:60]: Maybe a short comment?
```# Ensure transport solutions are not computed, nor stored.```

File: src/ott/solvers/linear/univariate.py [hunk 3]
```diff
@@ -261,36 +260,41 @@ def uniform_distance(
...
+    return_transport: whether to return mapped pairs.
 
   Returns:
-    optimal transport cost, a list of ``n+m`` paired indices, and their
+    optimal transport cost, a list of ``n + m`` paired indices, and their
     corresponding transport mass. Note that said mass can be null in some
     entries, but sums to 1.0
+
+  Notes:
+    Implementation inspired by ``scipy`` implementation for
```
Comment (marcocuturi) [inline src/ott/solvers/linear/univariate.py:295]: ```Inspired by ``scipy`` implementation ...``` to avoid double use of "implementation"

**Review by marcocuturi — APPROVED** [review]
thanks! minor comments

File: docs/neural/solvers.rst [hunk 4]
```diff
@@ -0,0 +1,24 @@
+ott.neural.solvers
+==================
+.. module:: ott.neural.solvers
+.. currentmodule:: ott.neural.solvers
+
+Under reconstruction. .. TODO(marcocuturi): add some nice text here please.
```
Comment (michalk8) [inline docs/neural/solvers.rst:6]: @marcocuturi also here, thank you!

File: docs/neural/index.rst [hunk 5]
```diff
@@ -0,0 +1,36 @@
+ott.neural
+==========
+.. module:: ott.neural
+.. currentmodule:: ott.neural
+
+Under reconstruction. .. TODO(marcocuturi): add some nice text here please.
```
[the remark this answers is not in the document]
Comment (marcocuturi) [reply docs/neural/index.rst:6]: wasnt sure you wanted it now :) 

In contrast to most methods presented in :mod:`~ott.solvers`, the goal of the :mod:`~ott.neural` module is to parameterize optimal transport maps and couplings as _neural networks_. These neural networks can _generalize to new samples_, in the sense that they can be conveniently evaluated outside training samples. This module implements layers, models and solvers to estimate such neural networks.

File: docs/neural/solvers.rst (hunk 4 above)
Comment (marcocuturi) [reply docs/neural/solvers.rst:6]: This module implements various solvers to estimate optimal transport between two probability measures, through samples, parameterized as neural networks. These neural networks are described in :mod:`~ott.neural.models`, borrowing lower-level components from :mod:`~ott.neural.layers`, building from the
`flax <https://flax.readthedocs.io/en/latest/examples.html>`__ library.

**Pull request merged**